src/sorting/quicksort-functional.js

```diff
@@ -0,0 +1,38 @@
+(function (exports) {
+
+  'use strict';
+
+  /**
+   * The quicksort algorithm (functional variant). It's complexity is O(nlog n).
+   *
+   * @public
+   */
+  var quickSort = (function () {
+
+    /**
+     * Sorts given array.
+     *
+     * @private
+     * @param {array} array Array which should be sorted
+     * @returns {array} array Sorted array
+     */
+    return function quickSort(array, left, right, cmp) {
```

Comment:
You're not really using `left`, `right` and `cmp`. Maybe it's a good idea to use `cmp` to allow more flexible sorting.